doc/src/guides/solving/solve-matrix-equation.md

```diff
@@ -0,0 +1,313 @@
+    ⎣1  0⎦
+>>> y = L.solve(P*b) # Step-by-step approach, step 1
+>>> y
+    ⎡0⎤
+    ⎢ ⎥
+    ⎣2⎦
+>>> U.solve(y) # Step-by-step approach, step 2
+    ⎡  2⋅e  ⎤
+    ⎢───────⎥
+    ⎢c⋅e + d⎥
+    ⎢       ⎥
+    ⎢   2   ⎥
+    ⎢───────⎥
+    ⎣c⋅e + d⎦
+>>> U.solve(L.solve(P*b)) # One-line approach
+    ⎡  2⋅e  ⎤
+    ⎢───────⎥
+    ⎢c⋅e + d⎥
+    ⎢       ⎥
+    ⎢   2   ⎥
+    ⎢───────⎥
+    ⎣c⋅e + d⎦
+>>> U.solve(L.solve(P*b2)) # Repeating one-line approach for b2
+    ⎡  4⋅e  ⎤
+    ⎢───────⎥
+    ⎢c⋅e + d⎥
+    ⎢       ⎥
+    ⎢   4   ⎥
+    ⎢───────⎥
+    ⎣c⋅e + d⎦
+```
+
+or compute the inverse matrix using
+{meth}`~sympy.matrices.matrices.MatrixBase.inv`:
+
+```py
+>>> from sympy import symbols, Matrix
+>>> c, d, e = symbols("c, d, e")
+>>> A = Matrix([[c,d], [1, -e]])
+>>> b = Matrix([2, 0])
+>>> b2 = Matrix([4, 0])
+>>> inv = A.inv()
+>>> inv
+    ⎡   e        d   ⎤
+    ⎢───────  ───────⎥
+    ⎢c⋅e + d  c⋅e + d⎥
+    ⎢                ⎥
+    ⎢   1       -c   ⎥
+    ⎢───────  ───────⎥
+    ⎣c⋅e + d  c⋅e + d⎦
+>>> inv * b # Solves to Ax = b for x
+    ⎡  2⋅e  ⎤
+    ⎢───────⎥
+    ⎢c⋅e + d⎥
+    ⎢       ⎥
+    ⎢   2   ⎥
+    ⎢───────⎥
+    ⎣c⋅e + d⎦
+>>> inv * b2 # Solves to Ax = b2 for x
+    ⎡  4⋅e  ⎤
+    ⎢───────⎥
+    ⎢c⋅e + d⎥
+    ⎢       ⎥
+    ⎢   4   ⎥
+    ⎢───────⎥
+    ⎣c⋅e + d⎦
+```
+
+### Work With Symbolic Matrices
+
+The computational complexity of manipulating symbolic matrices can increase
+rapidly with matrix size. For example, the number of terms in the determinant of
+a symbolic matrix increases with the factorial of the matrix dimension. As a
+result, the maximum dimensionality of matrices that can be solved is more
+limited than for numerical matrices. For example, the determinant of this 4x4
+symbolic matrix has 24 terms with four elements in each term:
+
+```py
+>>> from sympy import MatrixSymbol
+>>> A = MatrixSymbol('A', 4, 4).as_explicit()
+>>> A
+⎡A₀₀  A₀₁  A₀₂  A₀₃⎤
+⎢                  ⎥
+⎢A₁₀  A₁₁  A₁₂  A₁₃⎥
+⎢                  ⎥
+⎢A₂₀  A₂₁  A₂₂  A₂₃⎥
+⎢                  ⎥
+⎣A₃₀  A₃₁  A₃₂  A₃₃⎦
+>>> A.det()
+    A₀₀⋅A₁₁⋅A₂₂⋅A₃₃ - A₀₀⋅A₁₁⋅A₂₃⋅A₃₂ - A₀₀⋅A₁₂⋅A₂₁⋅A₃₃ + A₀₀⋅A₁₂⋅A₂₃⋅A₃₁ + A₀₀⋅A₁
+    ₃⋅A₂₁⋅A₃₂ - A₀₀⋅A₁₃⋅A₂₂⋅A₃₁ - A₀₁⋅A₁₀⋅A₂₂⋅A₃₃ + A₀₁⋅A₁₀⋅A₂₃⋅A₃₂ + A₀₁⋅A₁₂⋅A₂₀⋅
+    A₃₃ - A₀₁⋅A₁₂⋅A₂₃⋅A₃₀ - A₀₁⋅A₁₃⋅A₂₀⋅A₃₂ + A₀₁⋅A₁₃⋅A₂₂⋅A₃₀ + A₀₂⋅A₁₀⋅A₂₁⋅A₃₃ - 
+    A₀₂⋅A₁₀⋅A₂₃⋅A₃₁ - A₀₂⋅A₁₁⋅A₂₀⋅A₃₃ + A₀₂⋅A₁₁⋅A₂₃⋅A₃₀ + A₀₂⋅A₁₃⋅A₂₀⋅A₃₁ - A₀₂⋅A₁
+    ₃⋅A₂₁⋅A₃₀ - A₀₃⋅A₁₀⋅A₂₁⋅A₃₂ + A₀₃⋅A₁₀⋅A₂₂⋅A₃₁ + A₀₃⋅A₁₁⋅A₂₀⋅A₃₂ - A₀₃⋅A₁₁⋅A₂₂⋅
+    A₃₀ - A₀₃⋅A₁₂⋅A₂₀⋅A₃₁ + A₀₃⋅A₁₂⋅A₂₁⋅A₃₀
+```
+
+and solving a matrix equation of it takes about a minute, whereas the analogous
+3x3 matrix takes less than one second. The more unrelated, symbolic entries in a
+matrix, the more likely it is to be slow to manipulate.
+
+It is often much faster to solve as a system of linear
+[](solvers-algebraic-equations) rather than as a symbolic matrix equation for
+large matrix dimensions, so you may want to convert your matrix equation into a
```

Comment:
I did a test and it took <1 second to solve a 4x4 system of linear equations, versus more than a minute to solve the same problem as a matrix equation. If there are any caveats to this suggestion, please let me know.

Reply:
Reminding myself to replace this link with a link to "Solve a system of equations algebraically" page when #23802 is merged.

Reply:
The intention is that both methods would get faster in future.

Reply:
The point here is that the most general possible matrix equation, the inverse of a fully symbolic nxn matrix, is inherently complex because it involves the general nxn determinant which is n! terms. I've seen a lot of people try to do effectively this on the mailing list and StackOverflow. 

Of course, if the entries aren't fully symbolic, meaning they are related to each other somehow, or some of them are zero, then this isn't necessarily the case. 

By "fully symbolic" matrix, I mean something like `Matrix(10, 10, symbols('a:10:10'))`. In practice, the symbols may be called different things or even be larger expressions, but the point is that each entry is completely unrelated to the others, so no cancelation or simplification can occur when inverting it.